scripts/deploy-frontend.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/sh
+
+# source deploy-util.sh
+
+# The deployment strategy you wish to employ ( rolling update or setting up a new environment)
+DEPLOY_STRATEGY=${1}
+
+#The application name  defined via the manifest yml for the frontend
+CGHOSTNAME_FRONTEND=${2}
+CGHOSTNAME_BACKEND=${3}
+
+update_frontend()
+{
+    echo DEPLOY_STRATEGY: "$DEPLOY_STRATEGY"
+    echo FRONTEND_HOST: "$CGHOSTNAME_FRONTEND"
+    echo BACKEND_HOST: "$CGHOSTNAME_BACKEND"
+    cd tdrs-frontend || exit
+
+    echo "REACT_APP_BACKEND_URL=https://$CGHOSTNAME_BACKEND.app.cloud.gov/v1" >> .env.production
```

Comment:
i'd like to better understand what's happening here re: `.env.production` @jtwillis92

Reply:
In the context of the frontend app the `.env.production` file is used to provide build time overrides for settings committed to the repo in the `.env` files for the frontend. This is distinct from the one used for the backend in that it does not contain secrets.

For React the environment variables are evaluated and injected to the code at build time so the environment variables set on the Cloud.gov application instance have no impact on a running site. We have to do this prior to running `yarn build` so that the correct URLs are in place but those URLs change between different deployment targets so we dynamically assign it based on the supplied arguments to the script.

Reference docs: https://create-react-app.dev/docs/adding-custom-environment-variables/#what-other-env-files-can-be-used

Reply:
As an additional note - the environment variables set on the machine performing the `yarn build` step always override anything stored in any `.env` file, as mentioned in the docs linked above:
> These variables will act as the defaults if the machine does not explicitly set them.

See also this [further reference on dotenv](https://github.com/motdotla/dotenv#what-happens-to-environment-variables-that-were-already-set) - which is what is used by React to process these variables.

In the raft-tech CircleCI environment variables we had `REACT_APP_BACKEND_HOST` and `REACT_APP_BACKEND_URL` set to the localhost URL used for testing - which was overriding the build stage environment variables placed into `.env.production`. Since we already have a `.env.test` file configured for CI and checked in to the repo we do not need to define those in Circle CI at all - so they have since been removed.